Fix various EuiFormControlLayout usages

cee-chen opened the pull request:
## Summary

This is a follow up to EUI's Emotion conversion of **EuiFormControlLayout/Delimited** (see https://github.com/elastic/kibana/pull/190752, https://github.com/elastic/eui/pull/7954, and https://github.com/elastic/eui/pull/7957).

> [!note]
> Please manually QA your team's affected form control(s) to confirm they still look and behave as expected and are non-broken. The EUI team is not familiar enough with each plugin's setups to pull down and QA this PR ourselves.

While QA testing the upgrade, I noticed a few incorrect usages of **EuiFormControlLayout** but wanted to wait until after the upgrade to push out fixes (to prevent delaying the PR further). In general, here is EUI's [recommended usage of the component](https://eui.elastic.co/#/forms/form-controls#form-control-layout):

- Where possible, **simply don't use it**. Almost all form controls are **already** automatically wrapped in any EuiFormControlLayout by default, and should accept a large majority of the props that the layout accepts.
- If you **must** use it, set the `controlOnly` prop on the child input/control to avoid buggy styling (e.g. duplicate borders).
- If you can't do either of the above for any reason (e.g. missing prop support), reach out to the EUI team to ask for your UX as a feature request!

### Checklist

Delete any items that are not applicable to this PR.

- [x] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [x] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [x] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [x] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

cee-chen committed:
[EuiFormControlLayout/Delimited] Fix nested inputs missing `controlOnly` props

- prevents duplicate borders and other buggy effects

cee-chen committed:
[EuiDatePicker] Remove unnecessary `EuiFormControlLayout` wrappers

- EuiDatePicker supports `prepend` now, so move the prop directly onto the component

- EuiDatePicker also supports being cleared via `onClear`, no need to add a control layout for that button

cee-chen committed:
[examples] remove incorrect usage of `EuiFormControlLayout`

cee-chen committed:
[EuiFieldText] Remove unnecessary usages of `EuiFormControlLayout`

- the component already comes wrapped with EuiFormControlLayout by default, so we just need to move the props to the component itself

cee-chen committed:
[EuiFieldNumber] Remove unnecessary usages of `EuiFormControlLayout`

- the component already comes wrapped with EuiFormControlLayout by default, so we just need to move the props to the component itself

cee-chen committed:
[Triggers Actions UI] Remove unnecessary EuiFormControlLayout wrappers + fix empty label

- should be using `hasEmptyLabelSpace` prop instead of a space

- form controls already come wrapped with EuiFormControlLayout by default, so we just need to move the props to the component itself

cee-chen committed:
[EuiFieldSearch] Remove unnecessary `EuiFormControlLayout.onClear` wrapper

- field search components already support being cleared by default

- we need to move the onClear logic to the callbacks instead

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!

cee-chen committed:
Update snapshots

elasticmachine commented:
Pinging @elastic/eui-team (EUI)

yngrdyn reviewed (approved):
LGTM

elasticmachine commented:
Pinging @elastic/obs-ux-infra_services-team (Team:obs-ux-infra_services)

elasticmachine commented:
Pinging @elastic/obs-ux-management-team (Team:obs-ux-management)

adcoelho reviewed (approved):
Tested the `triggers_actions_ui` changed and everything still works as expected 👍 

Approved for `response-ops`

Heenawter reviewed (commented):
Book example embeddable + maps custom icon picker changes LGTM. Code review + tested locally to ensure that the HTML looks as expected - verified that the Maps `EuiFieldText` component is no longer wrapped in a double `EuiFormControlLayout`

cauemarcondes reviewed (approved):
LGTM

jeramysoucy commented on x-pack/packages/security/api_key_management/src/components/token_field.tsx:
This does not look like it is working as expected. The copy icon button is no longer rendering.

This PR:
<img width="532" alt="Screenshot 2024-09-13 at 6 05 36 PM" src="https://github.com/user-attachments/assets/c482d70c-acfd-49fa-95f1-80332d9d6e77">

Main:
<img width="511" alt="Screenshot 2024-09-13 at 6 13 27 PM" src="https://github.com/user-attachments/assets/b6a44cc9-d4f9-4b05-95a4-b37fd136b619">

cee-chen replied:
Let me know if this is working now with the latest commit! Thank you so much for checking this! 🙇

jeramysoucy replied:
The format drop-down and copy button now appear. The copy button is working, but the format selector is a bit strange. When "Encoded" is selected, it is highlighted in the list, but when "Beats" is selected, "Logstash" is highlighted in the list. When "Logstash" is selected, nothing is highlighted in the list. The format of the key for Beats and Logstash are identical, not sure if that is correct.

I have confirmed the exact same behavior in Main, so I will approve. Not sure if this is an issue with our code or the EUI components. I'll open an issue to look more closely.

jeramysoucy replied:
Issue: https://github.com/elastic/kibana/issues/193127

cee-chen replied:
I poked into this a bit locally and was able to fix the issue! Thankfully just a small oneliner. https://github.com/elastic/kibana/pull/192779/commits/661273c3e54895aaa1173b912d9a59f7ff500ee7

cee-chen commented on x-pack/packages/security/api_key_management/src/components/token_field.tsx:
```diff
@@ -29,12 +28,25 @@ export interface TokenFieldProps extends Omit<EuiFieldTextProps, 'append'> {
-    <EuiFormControlLayout
+    <EuiFieldText
+      data-test-subj="apiKeyTokenField"
+      controlOnly
```
Shoot, I left in this `controlOnly` prop by accident. Thanks so much for catching that Jeramy! 🙇 
```suggestion
```

cee-chen committed:
Fix API token field missing append/prepend

ryankeairns reviewed (approved):
SCSS check

cee-chen committed:
Merge branch 'main' into eui-form-layout-fixes

dominiqueclarke reviewed (approved):
Obs ux management changes LGTM

szwarckonrad reviewed (changes requested):
DW changes introduce regressions in functionality.


`main`

https://github.com/user-attachments/assets/38c38843-7ab4-4d8f-a31f-3ad0cabacc4d

`cee-chen:eui-form-layout-fixes`

https://github.com/user-attachments/assets/679c04cd-53fb-44db-aefc-91cb4ec535d0

jeramysoucy reviewed (approved):
Kibana security changes LGTM!

cee-chen committed:
Merge branch 'main' into eui-form-layout-fixes

rylnd reviewed (commented):
@cee-chen the Detection Engine changes look to be equivalent to `main` for `ScheduleItem`, (which doesn't look correct to me, but I think that's a separate issue:
 
<img width="473" alt="Screenshot 2024-09-17 at 10 25 54 PM" src="https://github.com/user-attachments/assets/3095e628-be81-4df0-aff1-2ec6a3e48c89">

), I'm seeing the following differences in `DurationInput` :

1. Extra border where previously there was none:
    <img width="473" alt="Screenshot 2024-09-17 at 10 30 19 PM" src="https://github.com/user-attachments/assets/215d0995-647f-422b-93a0-97fb3b161f16">
2. Disabling the input leads to an even more drastic change relative to the container:
    before: <img width="473" alt="Screenshot 2024-09-17 at 10 40 40 PM" src="https://github.com/user-attachments/assets/3896c217-a5f7-47d7-bfd8-79b8846cee45">
   after: <img width="473" alt="Screenshot 2024-09-17 at 10 30 14 PM" src="https://github.com/user-attachments/assets/9886d853-9571-4e1a-9654-ee773f179d34">
   
I'd be happy to pair if you need help here, please reach out on Slack.

cee-chen committed:
[PR feeedback] Fix defend workflows UX regression

- `searchValue` state is a step behind the search value string passed by EUI, so use that value instead to ensure we're not proceeding with `onSearch` split logic

cee-chen committed:
[PR feedback] Fix security rule schedule append+select styling

- a lot of this can be cleaned up

- `!important`s aren't ideal but are the fastest way forward here

cee-chen commented:
@szwarckonrad @rylnd Thank you both for the thorough QA and help with repro steps! I should have fixed both behaviors, but please let me know if not

**Security defend workflows:**

![defend](https://github.com/user-attachments/assets/9a3b5503-3690-4c2f-b4db-633c2a06c99d)

(ignore the lastpass autofill at the end, that's my browser/extension 😅)

**Security detection engine:**

<img width="323" alt="" src="https://github.com/user-attachments/assets/9586118e-a1f5-4165-827d-897b8d448659">
 
<img width="297" alt="" src="https://github.com/user-attachments/assets/3bc5aaac-35d2-442b-a7d1-31d00300e821">

cee-chen committed:
[bonus] Fix incorrect selection state on API key format dropdown

`* 2` index is unnecessary, EuiContextMenuPanel accounts for non-interactive elements

rylnd reviewed (approved):
@cee-chen thank you so much for fixing those issues. The Detection Engine changes look great!

szwarckonrad reviewed (approved):
No regression spotted with the latest changes. Thank you for addressing these! 🙌

cee-chen committed:
Merge branch 'main' into eui-form-layout-fixes

cee-chen committed:
Merge branch 'main' into eui-form-layout-fixes

cee-chen committed:
Merge branch 'main' into eui-form-layout-fixes

cee-chen committed:
Clean up Lens annotation datepicker UI further

- make the form element compressed to match the rest of the form controls in the panel

- remove unnecessary `EuiFormLabel` - one gets auto generated if we just pass a string

- move className override to parent and target the `__prepend` className

cee-chen commented:
@elastic/kibana-visualizations Once CI passes, I'll be requesting an admin merge from Kibana Ops. This PR contains several fixes for other teams that I don't want to hold up. I've pulled down and tested the UI owned by your team (Maps and Annotations) and confirm that they look and behave as expected:

### Annotations:

<img width="359" alt="" src="https://github.com/user-attachments/assets/41c986a4-ac71-4055-b7ae-d4bb63e5d265">
<img width="357" alt="" src="https://github.com/user-attachments/assets/163f0ebf-0307-4007-8c31-f9ae49770bca">

### Maps:

<img width="483" alt="" src="https://github.com/user-attachments/assets/7aad25e5-e8e1-4802-a7bd-bcdb42ed8234">
<img width="343" alt="" src="https://github.com/user-attachments/assets/4da81ce1-f0a2-4e98-8687-c01dfd96d2a0">

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/193925">](https://github.com/elastic/kibana/pull/193925)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->
